src/pop/blockchain/alt_block_tree.cpp

```diff
@@ -108,8 +108,40 @@ void AltBlockTree::acceptBlock(index_t& index,
                                ValidationState& state) {
   setPayloads(index, payloads);
 
-  if (index.pprev->isConnected()) {
-    connectBlock(index, state);
+  if (!index.pprev->isConnected()) {
+    return;
+  }
+
+  connectBlock(index, state);
+
+  // use non-recursive algorithm to connect the descendants.
+  // recursive algorithm caused segfaults on linux alpine because of deep
+  // recursion - thanks @daedalom for reporting it!
+  std::stack<BlockIndex<AltBlock>*> stack;
+  for (auto* pnext : index.pnext) {
+    stack.push(pnext);
+  }
+
+  while (!stack.empty()) {
+    auto* top = stack.top();
+    stack.pop();
+
+    // we never push nullptr to stack
+    VBK_ASSERT(top != nullptr);
+    if (!top->hasFlags(BLOCK_HAS_PAYLOADS)) {
+      // we can't connect this subtree because current block is 'in flight' -
+      // we never added payloads
+      continue;
+    }
+
+    // do connect
+    ValidationState dummy;
+    connectBlock(*top, dummy);
```

Comment:
Why not to use `state` variable, or save some info inside `state`

Reply:
`state` contains validation info about block in "acceptBlock". We don't want to clutter it with validation info from next blocks.